src/en/charms-fan.md

```diff
@@ -0,0 +1,202 @@
+Title: Juju and FAN networking
+TODO:  bug tracking: https://bugs.launchpad.net/juju/+bug/1733354 (remove constraints Note if no longer required)
+       hardcoded: default AWS instance type of 'm3.medium'
+       consider a third example: manually configure the FAN with Azure
+
+# Juju and FAN networking
+
+FAN networking addresses a need raised by the proliferation of container usage
+in an IPv4 context: the ability to manage the address space such that network
+connectivity among containers running on separate hosts is achieved.
+
+Juju integrates with the FAN to provide network connectivity between containers
+that was hitherto not possible. The typical use case is the seamless
+interaction between deployed applications running within LXD containers on
+separate Juju machines.
+
+## FAN overview
+
+The FAN is a mapping between a smaller IPv4 address space (e.g. a /16 network)
+and a larger one (e.g. a /8 network) where **subnets** from the larger one (the
+*underlay* network) are assigned to **addresses** on the smaller one (the
+*overlay* network). Connectivity between containers on the larger network is
+enabled in a simple and efficient manner.
+
+In the case of the above networks (/16 underlay and /8 overlay), each host
+address on the underlay "provides" 253 addresses on the overlay. FAN networking
+can thus be considered a form of "address expansion".
+
+Further reading on generic (non-Juju) FAN networking:
+
+ - [FAN networking][fan-ubuntu-wiki] : general user documentation
+ - [Container-to-Container Networking][fan-ubuntu-insights] : a less technical
+   overview
+ - [LXD network configuration][fan-lxd-config-options] : FAN configuration
+   options at the LXD level
+ - [`fanctl` man page][fan-fanctl-man-page] : configuration information at the
+   operating system level
+
+## Juju model FAN configuration
+
+Juju manages FAN networking at the model level and is enabled via the
+`container-networking-method` configuration option. This option can take on the
+following values:
+
+ - local : standard LXD; addressing based on the LXD bridge (e.g. lxdbr0)
+ - provider : addressing based on host bridge; works only with providers with
+   built-in container addressing support (e.g. MAAS with LXD)
+ - fan : FAN networking; works with any provider, in principle
+
+Once FAN is enabled, by setting the above option to 'fan', all that is needed
+is to map the underlay network to the overlay network. The `fan-config` model
+option is used for this and has the following format:
+
+  `<underlay-network>=<overlay-network>`
```

Comment:
It can be a space-separated list of multiple mappings: "<underlay-network1>=<overlay-network1> <underlay-network2>=<overlay-network2>"

Reply:
OK